src/inference/graph_inference.cpp

```diff
@@ -348,19 +378,24 @@ class GraphNodeAssignment : public Nice2Assignment {
 
   // Gets the score contributed by all arcs adjacent to a node.
   double GetNodeScore(const GraphInference& fweights, int node) const {
+    return GetNodeScoreGivenAssignment(fweights, node, assignments_);
+  }
+
+  // Gets the node score given an assignment
+  double GetNodeScoreGivenAssignment(const GraphInference& fweights, int node, const std::vector<Assignment> assignments) const {
     double sum = -GetNodePenalty(node);
-    const GraphInference::FeaturesMap& features = fweights.features_;
-    for (const GraphQuery::Arc& arc : query_->arcs_adjacent_to_node_[node]) {
-      GraphFeature feature(
-          assignments_[arc.node_a].label,
-          assignments_[arc.node_b].label,
-          arc.type);
-      auto feature_it = features.find(feature);
-      if (feature_it != features.end()) {
-        sum += feature_it->second.getValue();
-      }
-    }
-    return sum;
+        const GraphInference::FeaturesMap& features = fweights.features_;
```

Comment:
Nope, this gets the node score based on the assignment passed as argument. I use it when I compute the normalisation constant and I have to change the label of a certain node.